ethereum/permissioning/src/main/java/tech/pegasys/pantheon/ethereum/permissioning/NodeSmartContractPermissioningController.java

```diff
@@ -60,11 +66,30 @@ private static BytesValue hashSignature(final String signature) {
    *
    * @param contractAddress The address at which the permissioning smart contract resides
    * @param transactionSimulator A transaction simulator with attached blockchain and world state
+   * @param metricsSystem The metrics provider that is to be reported to
    */
   public NodeSmartContractPermissioningController(
-      final Address contractAddress, final TransactionSimulator transactionSimulator) {
+      final Address contractAddress,
+      final TransactionSimulator transactionSimulator,
+      final MetricsSystem metricsSystem) {
     this.contractAddress = contractAddress;
     this.transactionSimulator = transactionSimulator;
+
+    this.checkCounter =
```

Comment:
If we have the counter for permitted and unpermitted, do we really need the total counter or can we just infer from the other two metrics?
I'm not aware of the impact of the metrics in system resources so maybe we are better playing safe?

Reply:
There's an outcome where the permissioning check crashes (no contract, bad transaction etc) which increments the total counter but not the other 2. I'm assuming the metrics are no heavier than adding log lines so I decided to be distinct about all the things we might be checking.